src/themes/teams/components/Menu/menuItemStyles.ts

```diff
@@ -140,94 +166,108 @@ const menuItemStyles: IComponentPartStylesInput<IMenuItemProps, IMenuVariables>
-          }),
+          (vertical
+            ? pointing === 'end'
+              ? { borderRight: `${pxToRem(3)} solid ${v.typePrimaryActiveBorderColor}` }
+              : { borderLeft: `${pxToRem(3)} solid ${v.typePrimaryActiveBorderColor}` }
+            : pointingBeak({ props, variables: v, theme }))),
       }),
+
+      // focus styles
+      ...(isFromKeyboard && getActionStyles({ props, variables: v, color: v.defaultActiveColor })),
+
+      // hover styles
+      ':hover': getActionStyles({ props, variables: v, color: v.defaultActiveColor }),
     }
   },
 
-  anchor: ({ props, variables }): ICSSInJSStyle => {
-    const { active, iconOnly, pointing, type, underlined, vertical } = props
-    const { iconsMenuItemSize } = variables
+  anchor: ({ props, variables: v }): ICSSInJSStyle => {
+    const { active, iconOnly, isFromKeyboard, pointing, type, underlined, vertical } = props
 
     return {
       color: 'inherit',
       display: 'block',
+      cursor: 'pointer',
+
       ...(underlined
-        ? { padding: `0 0 ${pxToRem(8)} 0` }
+        ? { padding: `${pxToRem(4)} 0` }
         : pointing && vertical
           ? { padding: `${pxToRem(8)} ${pxToRem(18)}` }
           : { padding: `${pxToRem(14)} ${pxToRem(18)}` }),
-      cursor: 'pointer',
 
       ...(iconOnly && {
-        width: iconsMenuItemSize,
-        height: iconsMenuItemSize || '100%',
+        width: v.iconsMenuItemSize,
+        height: v.iconsMenuItemSize || '100%',
         padding: 0,
         display: 'flex',
         alignItems: 'center',
         justifyContent: 'center',
       }),
 
+      // active styles
+      ...(active &&
+        (type === 'primary'
+          ? {
+              ...(iconOnly && { color: v.typePrimaryActiveBorderColor }),
+
+              ...(underlined && {
+                color: v.typePrimaryActiveBorderColor,
+                ...underlinedItem(v.typePrimaryActiveBorderColor),
+              }),
+            }
+          : underlined && {
+              fontWeight: 700,
+              ...underlinedItem(v.defaultActiveColor),
+            })),
+
+      // focus styles
+      ...(isFromKeyboard && {
+        ...(type === 'primary'
+          ? {
+              ...(iconOnly && {
+                color: v.typePrimaryActiveBorderColor,
+                border: `1px solid ${v.typePrimaryActiveBorderColor}`,
+                borderRadius: v.circularRadius,
+              }),
+
+              ...(underlined && { color: v.typePrimaryActiveColor }),
+
+              ...(underlined && active && underlinedItem(v.typePrimaryActiveColor)),
+            }
+          : {
+              ...(iconOnly && {
+                border: `1px solid ${v.defaultActiveColor}`,
+                borderRadius: v.circularRadius,
+              }),
+
+              ...(underlined && { fontWeight: 700 }),
+
+              ...(underlined && active && underlinedItem(v.defaultActiveColor)),
+            }),
+      }),
+
+      ':focus': {
+        outline: 0,
```

Comment:
Agreed with @miroslavstastny we should still use outline: 0 for the variants of the Menu for which we don't provide some other styling.

Reply:
You should keep the outline for the examples that are not related to the Teams theme, or update it accordingly for the Teams Menu examples.

Reply:
I think there is a little misunderstanding here; All examples **do** have focus indicator visible and that's precisely the reason I took out the outline here at line 257. See DEMOs below for a basic vertical menu:

#### current master:
no focus indicator apart from the blue outline
![master](https://user-images.githubusercontent.com/5442794/46305858-edd7e600-c5b2-11e8-8a3f-e5e17d9bbe9d.gif)

#### this branch:
new focus indicator is the background color => outline is removed
every example has some sort of focus indicator; however, some examples have the same styling for both focused and active states which can be a little misleading
![new](https://user-images.githubusercontent.com/5442794/46305870-f4fef400-c5b2-11e8-8c29-c842f574766b.gif)

**tl;dr**
I don't thing we should do any more changes at this point

Reply:
@miroslavstastny @mnajdova @alinais

Reply:
I think this is enough for now, especially because these menus are not used in the teams theme. We can style them differently if we will receive more details spec about how they should look.